remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeFolderSizeRelation.java

```diff
@@ -0,0 +1,207 @@
+        int maxItems = params.getPaging().getMaxItems();
+        QName qName = nodeService.getType(nodeRef);
+
+        if (nodePerms != null && permissionService.hasPermission(nodeRef, PermissionService.READ) == AccessStatus.DENIED)
+        {
+            throw new AccessDeniedException("permissions.err_access_denied");
+        }
+
+        if(!"folder".equals(qName.getLocalName()))
+        {
+            throw new InvalidNodeTypeException(NOT_A_VALID_NODEID);
+        }
+
+        try
+        {
+            Action folderSizeAction = actionService.createAction(NodeSizeActionExecuter.NAME);
+            folderSizeAction.setTrackStatus(true);
+            folderSizeAction.setExecuteAsynchronously(true);
+            folderSizeAction.setParameterValue(NodeSizeActionExecuter.PAGE_SIZE, maxItems);
+            actionService.executeAction(folderSizeAction, nodeRef, false, true);
+            Map<String, Object> result = new HashMap<>();
+            result.put("executionId", nodeId);
+            return result;
+        }
+        catch (Exception ex)
+        {
+            LOG.error("Exception occured in NodeFolderSizeRelation:createById {}", ex.getMessage());
+        }
+        return null;
+    }
+
+    @Override
+    @WebApiDescription(title = "Returns Folder Node Size", description = "Returning a Folder Node Size")
+    @WebApiParameters({
+            @WebApiParam(name = "nodeId", title = "The unique id of the Node being addressed", description = "A single node id")})
+    public Map<String, Object> readById(String nodeId, String id, Parameters parameters)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        Node nodeInfo = nodes.getNode(nodeId);
+        NodePermissions nodePerms = nodeInfo.getPermissions();
+        QName qName = nodeService.getType(nodeRef);
+
+        if (nodePerms != null && permissionService.hasPermission(nodeRef, PermissionService.READ) == AccessStatus.DENIED)
+        {
+            throw new AccessDeniedException("permissions.err_access_denied");
+        }
+
+        if(!"folder".equals(qName.getLocalName()))
+        {
+            throw new InvalidNodeTypeException(NOT_A_VALID_NODEID);
+        }
+
+        try
+        {
+            Map<QName, Serializable> properties = nodeService.getProperties(nodeRef);
+            Map<String, Object> result = new HashMap<>();
+
+            if (properties == null || !properties.containsKey(FolderSizeModel.PROP_OUTPUT)) {
+                result.put("status", "NOT INITIATED");
+            } else {
+                String status = (String) properties.get(FolderSizeModel.PROP_STATUS);
+                if ("IN-PROGRESS".equals(status)) {
+                    result.put("status", status);
+                } else {
+                    Map<String, Object> mapResult = (Map<String, Object>) properties.get(FolderSizeModel.PROP_OUTPUT);
+                    mapResult.put("status", status);
+                    result = mapResult;
```

Comment:
we can return directly mapResult here right.

Reply:
No, we cannot return result map directly here, as we are keeping output in the same.
And instead of returning mapResult in every if-else we can return at the end.